test/helpers/all.test.js

```diff
@@ -127,10 +137,13 @@ describe('Filters', () => {
   });
 
   describe('.renderSpec', () => {
-    it('should work', async () => {
-      const doc = new AsyncAPIDocument({ asyncapi: '2.0.0', info: { title: 'dummy spec for testing', version: '1.5.34' } });
-
-      const result = renderSpec(doc);
+    it.skip('should work', async () => {
+      const schema = new AsyncAPIDocumentV2({
+        asyncapi: '2.0.0',
+        info: { title: 'test', version: '0.0.0' },
+        channels: {},
+      });
+      const result = renderSpec(schema);
```

Comment:
I unfortunately could not get the `renderSpec` part to work, which means that there will be no pre-rendering of the generated document (has to be loaded from the script).

Everything else works as expected.

Reply:
@magicmatatjahu if you have a hunch or know why do let me know 😆

Reply:
Why it doesn't work? Any error?

Reply:
It just renders empty string, no errors, nothing.

Reply:
Also, why are we doing both the pre-rendering and then rendering it again in the script 🤔

Reply:
If you use a generator, the generator has a separate reference to AsyncAPIDocumentV2 and html-template has another - template and the generator is run in separate processes as well as in separate contexts so the dependencies are completely different. Pass a normal specification (as string or object) - properly parsed - not a model.

Reply:
or even stringified spec

Reply:
> Also, why are we doing both the pre-rendering and then rendering it again in the script 🤔

For SEO - first we render pure html without actions/handlers for clicks action and then only hydrate html - better SEO, because we have html and more faster than normal "rendering" od React.

Reply:
> or even stringified spec

I tried both in the test (outside of generator) and it still renders `''`. Might need some longer debugging session to figure this out.

@magicmatatjahu do you see this as blocking the merge of this?

Reply:
> @magicmatatjahu do you see this as blocking the merge of this?

Reply:
> I tried both in the test (outside of generator) and it still renders ''. Might need some longer debugging session to figure this out.

But did you pass string or stringified spec? Also, maybe react component cannot recognize document inside https://github.com/asyncapi/asyncapi-react/blob/next/library/src/helpers/specification.ts#L15 this function

Reply:
Thats the most likely cause yea...  

> But did you pass string or stringified spec

Stringified spec using `stringify(asyncapi)` and json with `asyncapi.json()`.

Reply:
Tried to use the implementation with the generator and it seems to move it along.

But now the fun errors start 🤣 
```
Something went wrong:
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (/Users/lagoni/Documents/github/asyncapi-react/library/node_modules/react/cjs/react.development.js:1465:13)
    at useState (/Users/lagoni/Documents/github/asyncapi-react/library/node_modules/react/cjs/react.development.js:1496:20)
    at AsyncApiLayout (/Users/lagoni/Documents/github/asyncapi-react/library/src/containers/AsyncApi/Layout.tsx:28:61)
    at processChild (/Users/lagoni/Documents/github/html-template/node_modules/react-dom/cjs/react-dom-server.node.development.js:3353:14)
    at resolve (/Users/lagoni/Documents/github/html-template/node_modules/react-dom/cjs/react-dom-server.node.development.js:3270:5)
    at ReactDOMServerRenderer.render (/Users/lagoni/Documents/github/html-template/node_modules/react-dom/cjs/react-dom-server.node.development.js:3753:22)
    at ReactDOMServerRenderer.read (/Users/lagoni/Documents/github/html-template/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
    at Object.renderToString (/Users/lagoni/Documents/github/html-template/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
    at renderSpec (/Users/lagoni/Documents/github/html-template/helpers/all.js:161:25)
    at Index (/Users/lagoni/Documents/github/html-template/components/index.js:10:24)
```

Reply:
So you:
- have two versions of react in single run
- component is broken and it tries use hook when it should be able not to.

You have to debug.

Reply:
This might be because I am currently using symlink for the react components 🤔

Reply:
Yep, works like a charm, just the test I need to adapt.

Thanks @magicmatatjahu 🙇